server/src/test/java/org/opensearch/index/shard/IndexShardTests.java

```diff
@@ -2956,6 +2956,14 @@ public void testRestoreShardFromRemoteStore(boolean performFlush) throws IOExcep
             )
         );
 
+        // Make sure to drain refreshes from the shard. Otherwise, if the refresh is in-progress, it overlaps with
+        // deletion of segment files in the subsequent code block.
+        for (ReferenceManager.RefreshListener refreshListener : target.getEngine().config().getInternalRefreshListener()) {
+            if (refreshListener instanceof ReleasableRetryableRefreshListener) {
+                ((ReleasableRetryableRefreshListener) refreshListener).drainRefreshes();
+            }
+        }
+
```

Comment:
While this could just be an issue with this test, should we also check if we need to do this activity while closing shard in general?

Reply:
Yes, we need to check what are the possible issues that can occur and add it accordingly.